wp-content/plugins/acf-blocks-toolkit/includes/parts-kit.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * Add ACF Blocks to Parts Kit
+ *
+ * @package ACFBlocksToolkit
+ */
+
+use Viget\ACFBlocksToolkit\Block_Registration;
+
+add_filter(
+	'viget_parts_kit_block_%',
+	function ( string $output, string $block_name ): string {
+		$block = Block_Registration::get_block( $block_name );
+
+		if ( ! $block ) {
+			return $output;
+		}
+
+		if ( ! empty( $block['path'] ) ) {
+			$include = $block['path'] . '/block.php';
+
+			// Autoload block.php within block directory
+			if ( file_exists( $include ) ) {
+				require_once $include;
+			}
+		}
+
+		return acfbt_parse_inner_blocks( $output );
+	},
+	10,
+	2
+);
+
+/**
+ * Parse InnerBlocks Template
+ *
+ * @param string $output
+ *
+ * @return string
+ */
+function acfbt_parse_inner_blocks( string $output ): string {
+	// Check if InnerBlocks tag has a template attribute with regular expression
+	if ( ! preg_match( '/<InnerBlocks[^>]*template="([^"]*)"[^>]*>/i', $output, $matches ) ) {
+		return $output;
+	}
+
+	// Get the template attribute value
+	$template = $matches[1];
+	$template = htmlspecialchars_decode( $template );
+	$template = json_decode( $template, true );
+
+	$content = '';
+
+	foreach ( $template as $block_array ) {
+		$block = [
+			'blockName'   => $block_array[0],
+			'attrs'       => $block_array[1] ?? [],
+			'innerBlocks' => $block_array[2] ?? [],
+		];
+
+		$block    = acfbt_add_sample_data( $block );
+		$content .= apply_filters( 'the_content', trim( render_block( $block ) ) );
+		$content  = acfbt_parse_inner_blocks( $content );
+		$content  = acfbt_fill_empty_tags( $content );
+	}
+
+	$content = str_replace( '$', '\$', $content );
+
+	// Replace the InnerBlocks tag with the parsed content
+	return preg_replace( '/<InnerBlocks[^>]*\/>/i', $content, $output );
+}
+
+/**
+ * Get Sample Block properties
+ *
+ * @param array $block
+ *
+ * @return array
+ */
+function acfbt_get_sample_props( array $block ): array {
+	$props  = [];
+	$fields = get_fields( $block['name'] );
+
+	if ( ! is_array( $fields ) ) {
+		return $props;
+	}
+
+	foreach ( $fields as $field ) {
+		$props[ $field['name'] ] = acfbt_get_sample_data( $field );
+	}
+
+	return $props;
+}
+
+/**
+ * Return Sample Parts Kit Data
+ *
+ * @param array $field
+ *
+ * @return string|array
+ */
+function acfbt_get_sample_data( array $field ): string|array {
+	if ( in_array( $field['type'], [ 'text', 'select' ], true ) ) {
+		return 'Lorem Ipsum';
+	}
+
+	if ( 'image' === $field['type'] ) {
+		return [
+			'ID'          => 000,
+			'id'          => 000,
+			'title'       => 'placeholder',
```

Comment:
The previous iteration of `craft-viget`base` has some [nice helpers for generating random text](https://code.viget.com/craft-viget-base/parts-kit.html#components-exposed-to-twig-craftviget-sub-object) for the parts kit that might be cool to check out.

https://github.com/vigetlabs/craft-viget-base/blob/8def1735a0b3b8ef62661562dd01ee9eaf6fbdb1/src/services/PartsKit.php#L192